### ISSUE TITLE
Zero vector error from Pinecone while storing core memories 

### ISSUE THREAD

**Issue by chandanmaurya0** [langchain-ai/lang-memgpt#4]
I am trying to run the langgraph code locally using Pinecone , OpenAI Model, and embeddings. But I am getting below error from Pinecone. 

_**HTTP response body: {"code":3,"message":"Dense vectors must contain at least one non-zero value. Vector ID 'user/18682/core' contains only zeros","details":[]}**_

It seems Pinecone shouldn't support all zero vector documents. does anyone face similar kind of issue?

**Comment by Jaid844** [comment]
Yeah Problem with the empty vector seems like you cannot index it !
![Screenshot 2024-07-19 215950](https://github.com/user-attachments/assets/67dc38f5-3044-4b23-9eb2-37766457da24)


![Screenshot 2024-07-19 220143](https://github.com/user-attachments/assets/97fd0567-daac-4e1d-ad82-d6a11f57ac1a)

**Comment by hinthornw** [comment]
Oh good catch! Looks like I changed it too hastily

**Comment by Jaid844** [comment]
And the
 `constants.TYPE_KEY: "recall"`, in the store_core_memory tool should be 
`constants.TYPE_KEY: "core"` otherwise you wont filter out the core memory

**Comment by vidimitrov** [comment]
@hinthornw do you know what should be the fix for the `PineconeApiException`? I am also getting the `Dense vectors must contain at least one non-zero value` error...
